File: Searching/JavaCodes/Leetcode441.java
class Solution {
    public static int arrangeCoins(int manaDaggaraUnnaCoins) {
       int start=1,end=manaDaggaraUnnaCoins;
       if(manaDaggaraUnnaCoins==1)return 1;
       while(start<end){
           int stairs=end+(start-end)/2;
           long enniKavali=(long)stairs*(stairs+1)/2;
           if(enniKavali>manaDaggaraUnnaCoins){
               end=stairs-1;
           }
           else{
               start=stairs;
           }
       }
       return start;
    }
}

File: Searching/PythonCodes/Leetcode441.py
class Solution(object):
    def arrangeCoins(self, n):
        
        start, end = 1, n
        if n == 1:
            return 1
        while start < end:
            stairs = start + (end - start) // 2
            enniKavali = (stairs * (stairs + 1)) // 2
            if enniKavali > n:
                end = stairs
            else:
                start = stairs+1
        return end-1
